sequencer/src/api/sql.rs

```diff
@@ -648,15 +667,31 @@ async fn reward_header_dependencies(
             continue;
         }
 
-        let epoch = epoch_from_block_number(height, epoch_height);
+        let first_epoch = first_epoch.context("first epoch not found")?;
+
+        let epoch = EpochNumber::new(epoch_from_block_number(height, epoch_height));
```

Comment:
nit: `proposal_epoch`

Just because we have two `epoch` values we are comparing, both should have different names than just "epoch".